platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/builder/DependencyGraphBuilder.java

```diff
@@ -205,11 +216,19 @@ private void traverseGraph(final ResolveState resolveState) {
         }
     }
 
-    private static void registerCapabilities(final ResolveState resolveState, final NodeState node) {
-        CapabilitiesConflictHandler capabilitiesConflictHandler = resolveState.getConflictTracker().getCapabilitiesConflictHandler();
-        node.forEachCapability(capabilitiesConflictHandler, new Action<Capability>() {
+    /**
+     * Detect and register capability conflicts for the given node, deselecting the node if
+     * it is involved in a conflict.
+     *
+     * @return true iff a conflict was detected.
+     */
+    private static boolean registerCapabilities(final ResolveState resolveState, final NodeState node) {
+        AtomicBoolean foundConflict = new AtomicBoolean(false);
```

Comment:
You are correct to why `Atomic` is used. This is a common pattern to use `Atomic` in the case where you want a primitive pointer, but Java does not natively support it. 

In practice all this code needs to be reorganized, since it is a mess, but that is out of scope for this PR. 

I don't think a comment is necessary for this one

Reply:
It's just a comment though, it literally can't hurt.  Regardless of how common the practice is, the fact that reading this type in this particular code did cause me to think harder than I need to seems to be evidence for adding it. 🙂

Reply:
I've decided to not add a comment